oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/registration/ClientRegistration.java

```diff
@@ -153,6 +155,7 @@ public String toString() {
 		private String tokenUri;
 		private UserInfoEndpoint userInfoEndpoint = new UserInfoEndpoint();
 		private String jwkSetUri;
+		private Map<String, Object> configurationMetadata;
```

Comment:
I think it would be good to make this by default empty so user's don't need to worry about NPE

Reply:
NPE would never happen with `configurationMetadata` - see where it gets [init](https://github.com/spring-projects/spring-security/pull/5729/files#diff-804db4206dece0e20061c06b32546443R505)